internal/node/node.bzl

```diff
@@ -248,7 +248,9 @@ fi
     #
     # Rules such as nodejs_image should use only ctx.toolchains["@build_bazel_rules_nodejs//toolchains/node:toolchain_type"].nodeinfo
     # when building the image as that will reflect the selected --platform.
-    node_tool_files = ctx.files._node[:]
+    node_tool_files = ctx.files.node[:]
```

Comment:
The changes to node.bzl are unrelated to the basic e2e test case above right?

Lets do these in a separate PR where we add the `toolchain` attribute.

Reply:
this change is still needed to include nodejs_binary and nodejs_test in this pr where node can be specified or not. A follow up pr will address the toolchain changes. This allows a user to set the node to use but doesn't fully remove the toolchain yet. Might be better to address this in the follow up pr